src/components/forms/Combobox.tsx

```diff
@@ -549,6 +552,15 @@ export function ComboboxStatic<T = any>({
     };
   }, [initiallyVisible]);
 
+  useEffect(() => {
+    if (
+      comboboxInputRef &&
+      comboboxInputRef.current === document.activeElement
+    ) {
+      !isOpen && setIsOpen(true);
+    }
+  }, [comboboxInputRef.current]);
```

Comment:
@beganovich The Modal component, by default, focuses on the first focusable element in the entire modal. Therefore, once this happens in any part of the app, it will provide a much better UX if we open the dropdown popup (if the Combobox input field is focused).